### PR TITLE
Ogs script restructuring

### PR DESCRIPTION
**Pull request by jbathmann**
This PR is based on PR #154 .
It works just fine with all existing test cases. However, the existing ogs setups need to be replaced as done in the examples.
@mcwimm can you please check with your setups?

### CODE REVIEW

File: TreeModelLib/BelowgroundCompetition/NetworkOGSLargeScale3D/NetworkOGSLargeScale3D.py [hunk 1]
```diff
@@ -117,11 +117,3 @@ def calculateBelowgroundResources(self):
 
         # OGS stuff - update ogs parameters
         self.renameParameters()
-
```
[the remark this answers is not in the document]
Comment (jbathmann) [reply TreeModelLib/BelowgroundCompetition/NetworkOGSLargeScale3D/NetworkOGSLargeScale3D.py]: :+1: as discussed

File: TreeModelLib/BelowgroundCompetition/NetworkOGSLargeScale3DExternal/python_source.txt [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+# test/LargeTests/Test_Setups_large/ogs_example_setup/NetwOGS3D_SAZOI_BETTINA
+# .xml
+# In this example, 2 trees are placed in the center of a 20 x 10 m domain.
+# Contrary to OGSLargeScale3D, trees can establish root grafts and exchange
+# water through them.
+# Boundary conditions are defined on the left and right sight as 'Dirichlet'
+# boundary conditions for pressure and salinity concentration.
+# Tidal activity is not considered.
+
+seaward_salinity = 0.035
```
Comment (jbathmann) [inline TreeModelLib/BelowgroundCompetition/NetworkOGSLargeScale3DExternal/python_source.txt:15]: do you need this?

File: TreeModelLib/BelowgroundCompetition/NetworkOGSLargeScale3D/python_source.txt [hunk 3]
```diff
@@ -1,3 +1,6 @@
+
+
+
 ## This is an example for an OGS python boundary condition
 # The setup can be ran with:
 # test/LargeTests/Test_Setups_large/ogs_example_setup/NetwOGS3D_SAZOI_BETTINA
```
Comment (jbathmann) [inline TreeModelLib/BelowgroundCompetition/NetworkOGSLargeScale3D/python_source.txt:6]: where are the imports?

File: TreeModelLib/BelowgroundCompetition/OGSLargeScale3D/OGSLargeScale3D.py [hunk 4]
```diff
@@ -306,44 +317,43 @@ def copyPythonScript(self):
...
-
-            # OGS
-            if "constant_contributions.npy" in line:
-                line = line.replace("constant_contributions.npy",
-                                    constants_filename)
-            if "salinity_prefactors.npy" in line:
-                line = line.replace("salinity_prefactors.npy",
-                                    prefactors_filename)
+
+        if self._abiotic_drivers:
```
[the remark this answers is not in the document]
Comment (jbathmann) [reply TreeModelLib/BelowgroundCompetition/OGSLargeScale3D/OGSLargeScale3D.py]: :+1:

File: TreeModelLib/BelowgroundCompetition/OGSLargeScale3D/OGSLargeScale3D.py [hunk 5]
```diff
@@ -306,44 +317,43 @@ def copyPythonScript(self):
...
+            for abiotic_factor in self._abiotic_drivers.iterchildren():
+                line = (abiotic_factor.tag + " = " +
+                        abiotic_factor.text + "\n")
+                target.write(line)
+
+        # OGS
+        target.write(
```
[the remark this answers is not in the document]
Comment (jbathmann) [reply TreeModelLib/BelowgroundCompetition/OGSLargeScale3D/OGSLargeScale3D.py]: do you mean after rebase?

**Comment by mcwimm** [comment]
OGS and NetworkOGS test scripts work for me.

Should the user always provide a 'python_script' file?  In my case, for example, it's not required only for imports.
Should we move basic imports to python_source?

**Comment by jbathmann** [comment]
Compared to the previous implementation, the setup is more than 2.3 times faster than the previous implementation.
Key is the structure change in the cellfinder.

@mcwimm, does this structure work for your network setups?

**Comment by jvollhueter** [comment]
I think the structure has improved significantly with this PR. I have successfully tested the supplied example with Windows 11, Python 3.9, OGS 6.4.2 and could also calculate my own customized one based on the example setup.

**Review by jvollhueter — COMMENTED** [review]
I think the structure has improved significantly with this PR. I have successfully tested the supplied example with Windows 11, Python 3.9, OGS 6.4.2 and could also calculate my own customized one based on the example setup.

**Comment by mcwimm** [comment]
> Compared to the previous implementation, the setup is more than 2.3 times faster than the previous implementation. Key is the structure change in the cellfinder.
> 
> @mcwimm, does this structure work for your network setups?

I tested only the functionality (i.e. results not checked). Works for OGS-Network setup. External script were not tested.

File: TreeModelLib/BelowgroundCompetition/NetworkOGSLargeScale3D/NetworkOGSLargeScale3D.py [hunk 6]
```diff
@@ -49,7 +49,9 @@ def addTree(self, tree):
         # OGS stuff
         x, y = tree.getPosition()
         # add cell IDs and cell volume to tree
-        super().addCellCharateristics(x, y)
+        geometry = tree.getGeometry()
+        root_radius = geometry["r_root"]
```
Comment (jbathmann) [inline TreeModelLib/BelowgroundCompetition/NetworkOGSLargeScale3D/NetworkOGSLargeScale3D.py:53]: :+1:

File: TreeModelLib/BelowgroundCompetition/OGSLargeScale3DExternal/OGSLargeScale3DExternal.py [hunk 7]
```diff
@@ -56,7 +56,8 @@ def addTree(self, tree):
         parameter = tree.getParameter()
 
         # Cells affected by tree water uptake
-        super().addCellCharateristics(x, y)
+        root_radius = geometry["r_root"]
```
Comment (jbathmann) [inline TreeModelLib/BelowgroundCompetition/OGSLargeScale3DExternal/OGSLargeScale3DExternal.py:59]: :+1:
